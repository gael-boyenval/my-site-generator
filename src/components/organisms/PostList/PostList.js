import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Container from '../../layout/Container';
import { Margin, Padding } from '../../layout/Spaces';
import Grid from '../../layout/Grid';
import PostItem from '../../molecules/PostItem';
import { SubTitle } from '../../atoms/Typography';
import ButtonLink from '../../atoms/ButtonLink';

class PostList extends PureComponent {
  static propTypes = {
    posts: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  };

  render() {
    const { posts } = this.props;

    return (
      <Container>
        <Padding vertical="wide">
          <Margin bottom="xxxl">
            <Grid center>
              <Grid.Cell>
                <SubTitle>Mes derniers articles :</SubTitle>
              </Grid.Cell>
              <Grid.Cell width="fill">
                <Margin top="s" left="m">
                  <ButtonLink href="">voir le blog</ButtonLink>
                </Margin>
              </Grid.Cell>
            </Grid>
          </Margin>
          <Grid gutter>
            {posts.map(({ node }) => (
              <Grid.Cell key={node.fields.slug} width={1 / 3}>
                <Margin bottom="xxl">
                  <PostItem post={node} />
                </Margin>
              </Grid.Cell>
            ))}
          </Grid>
          <Margin bottom="xxl">
            <ButtonLink href="">voir le blog</ButtonLink>
          </Margin>
        </Padding>
      </Container>
    );
  }
}

export default PostList;
