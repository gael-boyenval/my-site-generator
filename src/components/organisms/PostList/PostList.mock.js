export default [
  {
    node: {
      fields: {
        slug: 'cest-un-hash-d-url-1',
      },
      frontmatter: {
        date: '2017-03-02',
        title: 'c’est un super titre 1',
      },
    },
  },
  {
    node: {
      fields: {
        slug: 'cest-un-hash-d-url-2',
      },
      frontmatter: {
        date: '2018-05-03',
        title: 'c’est un super titre',
      },
    },
  },
  {
    node: {
      fields: {
        slug: 'cest-un-hash-d-url-3',
      },
      frontmatter: {
        date: '2014-02-26',
        title: 'c’est un super titre 3',
      },
    },
  },
];
