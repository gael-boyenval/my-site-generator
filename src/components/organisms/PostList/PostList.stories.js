/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { withKnobs } from '@storybook/addon-knobs';

import PostList from './PostList';
import Mock from './PostList.mock';

storiesOf('04 - Organisms', module)
  .addDecorator(withKnobs)
  .add('PostList', withInfo()(() => <PostList posts={Mock} />));
