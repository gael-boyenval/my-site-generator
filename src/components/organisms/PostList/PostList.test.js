/* eslint-disable no-undef */
import React from 'react';
import PostList from './PostList';

const posts = { edges: [] };

describe('Component PostList', () => {
  it('should be render without crash', () => {
    const component = shallow(<PostList posts={posts} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
