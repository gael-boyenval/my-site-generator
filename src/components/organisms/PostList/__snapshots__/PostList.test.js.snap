// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PostList should be render without crash 1`] = `
<styled.div>
  <styled.div
    all={false}
    bottom="xxxl"
    horizontal={false}
    inline={false}
    left={false}
    right={false}
    top={false}
    vertical={false}
  >
    <styled.div
      center={true}
    >
      <styled.div>
        <styled.h2>
          Mes derniers articles :
        </styled.h2>
      </styled.div>
      <styled.div
        width="fill"
      >
        <styled.div
          all={false}
          bottom={false}
          horizontal={false}
          inline={false}
          left="m"
          right={false}
          top="s"
          vertical={false}
        >
          <ButtonLink
            href=""
          >
            voir le blog
          </ButtonLink>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div
    gutter={true}
  />
  <styled.div
    all={false}
    bottom="xxl"
    horizontal={false}
    inline={false}
    left={false}
    right={false}
    top={false}
    vertical={false}
  >
    <ButtonLink
      href=""
    >
      voir le blog
    </ButtonLink>
  </styled.div>
</styled.div>
`;
